src/vst/ivstchannelcontextinfo.rs

```diff
@@ -0,0 +1,48 @@
+use vst3_com::interfaces::IUnknown;
+use vst3_com::{c_void, com_interface};
+
+use crate::base::tresult;
+use crate::vst::{CString, ColorSpec};
+
+#[com_interface("0F194781-8D98-4ADA-BBA0-C1EFC011D8D0")]
+pub trait IInfoListener: IUnknown {
+    unsafe fn set_channel_context_infos(&mut self, list: *mut c_void) -> tresult;
+}
+
+pub enum ChannelPluginLocation {
+    kPreVolumeFader = 0,
+    kPostVolumeFader = 1,
+    kUsedAsPanner = 2,
+}
+
+pub type ColorComponent = u8;
+
+pub fn get_blue(cs: ColorSpec) -> ColorComponent {
+    (cs & 0x0000_00FF) as ColorComponent
+}
+
+pub fn get_green(cs: ColorSpec) -> ColorComponent {
+    ((cs >> 8) & 0x0000_00FF) as ColorComponent
+}
+
+pub fn get_red(cs: ColorSpec) -> ColorComponent {
+    ((cs >> 16) & 0x0000_00FF) as ColorComponent
+}
+
+pub fn get_alpha(cs: ColorSpec) -> ColorComponent {
+    ((cs >> 24) & 0x0000_00FF) as ColorComponent
+}
+
+pub const kChannelUIDKey: CString = b"channel uid".as_ptr() as *const _;
```

Comment:
Is this correct? The SDK documents these strings as `TChar`, which are UTF-16.

Reply:
Are you reading the code comments or the code itself in the SDK?

Reply:
`base/ftypes.h` I see that CString is a UTF16 string (const int16_t*) so what I'm asking is if these constants are valid with that pointer cast?

Reply:
Sadly, the one used here is a different CString, from `vst/vsttypes.h`. These types of name collisions are not nice, but that's the way it is.

Reply:
Gross. iirc byte string literals aren't null terminated in Rust, so these need the '\0' character at the end as well.

Reply:
Yeah, I forgot, I'll have to fix the other CString usages in the crate because they don't seem to have it either